File: aoc2015/days/day02.go
package days

import ("fmt"
		"bufio"
		"io"
		"os"
		"strings"
		"strconv"
		"aoc2015/common"
)

func Day02() {

	file, err := os.Open("input02.txt")
	if err != nil {
        return
    }
	defer file.Close()
	
	reader := bufio.NewReader(file)

	var sum int64 = 0
	var ribbon int64 = 0

	for {
		line, _, err := reader.ReadLine()

		if err == io.EOF {
				break
		}

		numberss := strings.Split(string(line), "x")

		var n []int64
		var ssum int64 = 0

		for i := range numberss {
			v, _ := strconv.ParseInt(numberss[i], 10, 64)
			n = append(n, v)
		}

		var ms int64 = 100000000000
		var mc int64 = 100000000000

		for i := 0; i <= 2; i++ {
			for j := i+1; j <= 2; j++ {
				side := n[i]*n[j]
				ms = common.Min(side, ms)
				mc = common.Min(2*n[i]+2*n[j], mc)
				ssum += 2*side
			}
		}

		sum += ssum + ms
		ribbon += mc + (n[0]*n[1]*n[2])
	}

	fmt.Println("Part 1:", sum)
 	fmt.Println("Part 2:", ribbon)


}

File: aoc2017.py/day02.py

text_file = open("input02.txt", "r")

lines = [x.rstrip("\n\r") for x in text_file.readlines()]

result1 = 0

for l in lines:
    li = [int(x) for x in l.split('\t')]
    result1 += abs(min(li)-max(li))

print("Part 1", result1)

result2 = 0

for l in lines:
    li = sorted([int(x) for x in l.split('\t')], reverse=True)
    
    for i in range(len(li)-1):
        for j in range(i+1, len(li)):
            if li[i]//li[j]*li[j] == li[i]:
                result2 += li[i]//li[j]

print("Part 2", result2)
